resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/ExceptionMapping.java

```diff
@@ -40,9 +40,13 @@ public void mapException(final Method method) throws Exception {
                 if (exception instanceof RuntimeException) throw (RuntimeException) exception;
                 if (exception instanceof Error) throw (Error) exception;
                 for (Class exc : method.getExceptionTypes()) {
-                    if (exc.isAssignableFrom(exception.getClass())) throw (Exception) exception;
+                    if (exception != null && exc.isAssignableFrom(exception.getClass())) {
```

Comment:
Does this ever return an array with a null value? I'm just curious because that seems weird if it does.

Reply:
This PR is to properly respond to an edge case inferred by the microprofile-rest-client 2.0 specification.  The spec defines interface ResponseExceptionMapper,
`This mapper will take a Response object retrieved via an invocation of a client 
  and convert it to a Throwable, if applicable.`

The ResponseExceptionMapper method toThrowable(Response),
`... may return null if no throwable should be raised.`

The user may provide custom implementations of this interface to handle exceptions.
A provider must provide an implementation of the interface
`Each implementation will provide out of the box a ResponseExceptionMapper implementation`

The default mapper can be disabled.
`This behavior can be disabled by adding a configuration property 
  microprofile.rest.client.disable.default.mapper with value true`

There can be a scenario in which the implementor's default mapper has been disabled
and the user's mapper implementation returns NULL.  The spec makes no statement about
handling of this scenario.  I can only infer that NULL is to be returned to the caller.

We must allow all the mapper "candidates" to be processed lines 38-47 in ExceptionMapping,
thus providing every opportunity for a mapper to handle the issue.  
A return value of NULL must be checked for at line 43 in ExceptionMapping before
exc.isAssignableFrom(exception.getClass()) is executed.  If this is not done
exc.isAssignableFrom throws an invocation exception because NULL.getClass()
is not valid; that stops further processing of "candidates" and the user's
mapper response is superseded by a misleading exception.

The sole function of ExceptionMapping.HandlerException is to process and throw
an exception.  It is only used in an exception handling block.  It is only
used for microprofile-rest-client proxy handling.  I think it is safe to
assume if the call to method mapException returns normally a user's custom
mapper has return NULL.

Reply:
I apologize this `null` check is absolutely correct otherwise `exception.getClass()` could throw an NPE.

It's the returning `null` that really concerns me. Without an `ResponseExceptionMapper` the error may be swallowed so someone getting an NPE because it returns `null` would have a hard time finding out what happened. We should probably at least log a debug message.

TBH though I don't have a better idea than `null` so it might be the best solution.

At least for the default though we should be throwing a `WebApplicationException` if we don't find a mapper.
> Each implementation will provide out of the box a ResponseExceptionMapper implementation that will map the response into a WebApplicationException whenever the response status code is >= 400.

Reply:
> We should probably at least log a debug message.

That seems like a good idea.

> At least for the default though we should be throwing a WebApplicationException if we don't find a mapper.

I don't think that would be valid because the spec also states that the user can
disable the default mapper.

> This behavior can be disabled by adding a configuration property microprofile.rest.client.disable.default.mapper with value true

Perhaps adding a log message that the default mapper has been disabled will suffice.

Reply:
I've added the proposed log messages

Reply:
I am once again an idiot, sorry @rsearls. I was thinking the `ExceptionMapping` was the default mapper. It definitely is not, I'm sorry.

Maybe what we do is log warning a message before returning `null` that the exception wasn't handled and see if we can log some kind of information about it. There really isn't anything better than returning `null` with the exception of raising an exception, but it would take effort to disable raising an exception so it seems it would be intentional.

Reply:
lol you already do that, but could we do something like `cause.getCause()` and if not `null` log either the full exception or just the `cause.getCause().getLocalizedMessage();`?

Reply:
I moved the log msg in ExceptionMapper so I could get access to the response object status code and message.